plugins/inputs/execd/execd.go

```diff
@@ -90,59 +97,62 @@ func (e *Execd) Stop() {
 	e.wg.Wait()
 }
 
-func (e *Execd) cmdLoop(ctx context.Context) {
+// cmdLoop watches an already running process, restarting it when appropriate.
+func (e *Execd) cmdLoop(ctx context.Context) error {
 	for {
 		// Use a buffered channel to ensure goroutine below can exit
 		// if `ctx.Done` is selected and nothing reads on `done` anymore
 		done := make(chan error, 1)
 		go func() {
-			done <- e.cmdRun()
+			done <- e.cmdWait()
 		}()
 
 		select {
 		case <-ctx.Done():
-			e.stdin.Close()
-			// Immediately exit process but with a graceful shutdown
-			// period before killing
-			internal.WaitTimeout(e.cmd, 200*time.Millisecond)
-			return
+			if e.stdin != nil {
+				e.stdin.Close()
+				// Immediately exit process but with a graceful shutdown
+				// period before killing
+				internal.WaitTimeout(e.cmd, 200*time.Millisecond)
+			}
+			return nil
 		case err := <-done:
 			log.Printf("E! [inputs.execd] Process %s terminated: %s", e.Command, err)
+			return err
 		}
 
-		log.Printf("E! [inputs.execd] Restarting in %s...", e.RestartDelay.Duration)
+		log.Printf("E! [inputs.execd] Restarting in %s...", time.Duration(e.RestartDelay))
 
 		select {
 		case <-ctx.Done():
-			return
-		case <-time.After(e.RestartDelay.Duration):
+			return nil
+		case <-time.After(time.Duration(e.RestartDelay)):
 			// Continue the loop and restart the process
+			if err := e.cmdStart(); err != nil {
+				return err
+			}
 		}
 	}
 }
 
-func (e *Execd) cmdRun() error {
-	var wg sync.WaitGroup
-
+func (e *Execd) cmdStart() (err error) {
 	if len(e.Command) > 1 {
 		e.cmd = exec.Command(e.Command[0], e.Command[1:]...)
 	} else {
 		e.cmd = exec.Command(e.Command[0])
 	}
 
-	stdin, err := e.cmd.StdinPipe()
+	e.stdin, err = e.cmd.StdinPipe()
 	if err != nil {
 		return fmt.Errorf("E! [inputs.execd] Error opening stdin pipe: %s", err)
```

Comment:
Can you fix this preexisting issue with the error strings in this function, they should be like: "error opening stdin pipe: %v", without the logging prefix.